### PR TITLE
Path: Avoid dictionary.keys() if possible

### PR DESCRIPTION
**Pull request by Roy-043**
See #6015.

- [x]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

---

### CODE REVIEW

File: src/Mod/Path/Path/Dressup/DogboneII.py [hunk 1]
```diff
@@ -184,10 +184,10 @@ def boneTip(self):
         return self.bone.tip()
 
     def boneIDs(self):
-        return list(sorted(self.bones.keys()))
+        return sorted(self.bones)
```
Comment (mlampert) [inline src/Mod/Path/Path/Dressup/DogboneII.py:187]: This changes the return type of the function. What's the rational for this change?

**Comment by Roy-043** [comment]
It is my understanding that the return type of the function does not change with this PR:
```
>>> my_dictionary = {"a": 1, "b": 2, "c": 3}
>>> type(sorted(my_dictionary))
<class 'list'>
>>> list(sorted(my_dictionary.keys())) == sorted(my_dictionary)
True
```

**Comment by mlampert** [comment]
you should try `type(list(...)) == type(....)`

**Comment by Roy-043** [comment]
Do you mean this?:
```
>>> type(list(sorted(my_dictionary.keys()))) == type(sorted(my_dictionary))
True
```

**Comment by mlampert** [comment]
look at that, it seems my understanding of `sorted` is outdated. Thanks for clearing that up.

**Pull request merged**